clients/input_copy.go

```diff
@@ -63,8 +70,62 @@ func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVi
 		}
 	}
 
+	if encryptedKey != "" {
+		c, e := getFile(context.Background(), requestID, inputFile.String())
+
+		if e != nil {
+			glog.Errorf("error getting file: %w", err)
+			return
+		}
+
+		var decryptionKey *rsa.PrivateKey
+		if decryptionKey, err = crypto.LoadPrivateKey(VodDecryptPrivateKey); err != nil {
+			glog.Errorf("error loading private key: %w", err)
+			return
+		}
+
+		if decryptedFile, err = crypto.DecryptAESCBC(c, decryptionKey, encryptedKey); err != nil {
+			glog.Errorf("error decrypting file: %w", err)
+			return
+		}
+	}
+
+	if decryptedFile != nil {
+		var size int64
+		decryptedFileUrl := osTransferURL.String()
+
+		log.Log(requestID, "Copying decrypted file to S3", "source", inputFile.String(), "dest", decryptedFileUrl)
+		size, err = CopyReaderFile(context.Background(), decryptedFile, decryptedFileUrl, "", requestID)
+		if err != nil {
+			err = fmt.Errorf("failed to copy file(s): %w", err)
+			return
+		}
+		log.Log(requestID, "Copied", "bytes", size, "source", inputFile.String(), "dest", decryptedFileUrl)
+
+		signedURL, err = SignURL(osTransferURL)
+		if err != nil {
+			return
+		}
+	}
+
+	if !isDirectUpload(inputFile) {
```

Comment:
This condition is not correct since the file may have already been copied above. In case of non-direct uploads (most of the encryption use case) we will copy the decrypted file to S3, then copy it again, but encrypted. Overriding the decryption work.

I think the ideal fix here would be moving the decryption work to this code block below and have a simple condition here 
`!isDirectUpload(inputFile) || encryptionKey != ""`